examples/jsm/csm/CSMShadowNode.js

```diff
@@ -0,0 +1,435 @@
+import {
+	Vector2,
+	Vector3,
+	MathUtils,
+	Matrix4,
+	Box3,
+	Object3D,
+	WebGLCoordinateSystem
+} from 'three';
+
+import { CSMFrustum } from './CSMFrustum.js';
+import { reference, uniform, int, float, vec4, vec2, If, Fn, min, renderGroup, positionView, Node, NodeUpdateType, shadow } from 'three/tsl';
+
+const _cameraToLightMatrix = new Matrix4();
+const _lightSpaceFrustum = new CSMFrustum();
```

Comment:
I think you must create `CSMFrustum` depending on the backend. Otherwise the result is incorrect for WebGL since the internal vertices are not correctly configured.